includes/handle-redirects.php

```diff
@@ -88,6 +89,12 @@ function get_redirect_uri( $url ) {
 	// If the URL is only a path, prefix it with the `home_url()`.
 	$to_url = Utilities\prefix_path( $to_url );
 
+	// Re-append any existing query string parameters.
+	if ( ! empty( $query_string ) ) {
+		$to_url .= strpos( $to_url, '?' ) !== false ? '&' : '?';
```

Comment:
I don't know if `$to_url` can have a non-empty hash at this point, but if it does, this will not work as expected.

Reply:
Yeah it's up to the individual added the redirect target 👍  Will update.

Reply:
Updated to support redirecting to and preserving URL fragments.